spec/requests/presence_controller_spec.rb

```diff
@@ -122,58 +122,69 @@
     let(:user3) { Fabricate(:user) }
 
     it "works" do
-      get "/presence/get", params: { channel: ch1.name }
+      get "/presence/get", params: { channels: [ch1.name] }
       expect(response.status).to eq(200)
-      body = response.parsed_body
-      expect(body["users"]).to eq([])
-      expect(body["count"]).to eq(0)
-      expect(body["last_message_id"]).to eq(MessageBus.last_id(ch1.message_bus_channel_name))
+      expect(response.parsed_body).to eq(
+        ch1.name => {
+          "users" => [],
+          "count" => 0,
+          "last_message_id" => MessageBus.last_id(ch1.message_bus_channel_name)
+        }
+      )
 
       ch1.present(user_id: user.id, client_id: SecureRandom.hex)
       ch1.present(user_id: user2.id, client_id: SecureRandom.hex)
       ch1.present(user_id: user3.id, client_id: SecureRandom.hex)
 
-      get "/presence/get", params: { channel: ch1.name }
-      body = response.parsed_body
-      expect(body["users"].map { |u| u["id"] }).to contain_exactly(user.id, user2.id, user3.id)
-      expect(body["users"][0].keys).to contain_exactly("avatar_template", "id", "name", "username")
-      expect(body["count"]).to eq(3)
-      expect(body["last_message_id"]).to eq(MessageBus.last_id(ch1.message_bus_channel_name))
+      get "/presence/get", params: { channels: [ch1.name] }
+      expect(response.status).to eq(200)
+      state = response.parsed_body[ch1.name]
+      expect(state["users"].map { |u| u["id"] }).to contain_exactly(user.id, user2.id, user3.id)
+      expect(state["users"][0].keys).to contain_exactly("avatar_template", "id", "name", "username")
+      expect(state["count"]).to eq(3)
+      expect(state["last_message_id"]).to eq(MessageBus.last_id(ch1.message_bus_channel_name))
     end
 
     it "respects the existence/security of the channel" do
       sign_in user
 
-      get "/presence/get", params: { channel: ch1.name }
+      get "/presence/get", params: { channels: [ch1.name] }
       expect(response.status).to eq(200)
+      expect(response.parsed_body[ch1.name]).not_to eq(nil)
 
-      get "/presence/get", params: { channel: secure_user_channel.name }
-      expect(response.status).to eq(404)
+      get "/presence/get", params: { channels: [secure_user_channel.name] }
+      expect(response.status).to eq(200)
+      expect(response.parsed_body).to eq({ secure_user_channel.name => nil })
```

Comment:
Not very sure here but I'm wondering why we need to return the channel name if the user is not allowed to view it.

Reply:
The idea is that `"/channel-name": null` means "this channel does not exist". The client-side interprets this, and throws a `PresenceChannelNotFound` error. 

Previously "does not exist" was indicated by a 404 error, but that's no longer possible because multiple channels are returned in a single API call.